iroha/src/sumeragi.rs

```diff
@@ -65,7 +65,7 @@ pub struct Sumeragi {
     /// Number of view changes after the previous block was committed
     number_of_view_changes: u32,
     invalidated_blocks_hashes: Vec<Hash>,
-    permissions_validator: PermissionsValidatorBox,
+    permissions_validator: Arc<PermissionsValidatorBox>,
```

Comment:
Box in a box?

Reply:
It is an Atomic Reference that wraps Box

Reply:
From the design standpoint PermissionValidatorBox is used inside the permission validators themselves, they should not be aware of the fact that they are used in async contexts

Reply:
Going through 2 references is very painfull. May be we can overcome this somehow? for example make permissions validator box always be Arc instead of Box?

Reply:
It is only in one place, where Arc is needed, but Box instead is used in all the internal nested structure. I don't think this will be a huge performance improvement if we change it anyway.